### PR TITLE
SLIS article migration

### LINKED ISSUE #3169: SLIS: Build article migration
# Motivation

The School of Library & Information Science website has about 300+ news articles on it. As part of the process of moving their site to D9, I would like to build a migration for this content and a migration to set up URL redirects from the old path to the new auto-generated one.

# Proposed solution

- [ ] Build News => Article migration to handle the following: 
  - Body field with inline images converted to media embeds.
  - `field_image_attach` (Image) field migrated into `field_image`.
- [ ] Build migration to set up redirect records from old path to new auto-generated one.

### PR DESCRIPTION
**Pull request by pyrello**
Resolves #3169 

# How to test

Code review is fine for this.

### CODE REVIEW

File: docroot/modules/custom/sitenow_migrate/sitenow_migrate.install [hunk 1]
```diff
@@ -41,3 +42,47 @@ function sitenow_migrate_install() {
     }
   }
 }
+
+/**
+ * Implements hook_install().
+ */
+function sitenow_migrate_migration_install($sitename) {
```
Comment (richardbporter) [inline docroot/modules/custom/sitenow_migrate/sitenow_migrate.install:49]: Docblock says `hook_install` but this looks like `hook_migration_install` - is this intended?

File: docroot/modules/custom/sitenow_migrate/sitenow_migrate.install [hunk 2]
```diff
@@ -41,3 +42,47 @@ function sitenow_migrate_install() {
...
+  $config_factory = \Drupal::configFactory();
+  $config = $config_factory->getEditable("config_split.config_split.${sitename}_migrate");
+  $config->set('status', TRUE);
+  $config->save(TRUE);
+}
+
+/**
+ * Implements hook_uninstall().
+ */
+function sitenow_migrate_migration_uninstall($sitename) {
```
Comment (richardbporter) [inline docroot/modules/custom/sitenow_migrate/sitenow_migrate.install:75]: Same thing here.

File: docroot/modules/custom/sitenow_migrate/sitenow_migrate.install (hunk 1 above)
Comment (pyrello) [reply docroot/modules/custom/sitenow_migrate/sitenow_migrate.install:49]: This is a helper function. There is no `hook_migration_install` function.

File: docroot/modules/custom/sitenow_migrate/sitenow_migrate.install (hunk 2 above)
Comment (pyrello) [reply docroot/modules/custom/sitenow_migrate/sitenow_migrate.install:75]: Ditto.

**Comment by pyrello** [comment]
Thanks @GaryRidgway for pointing out the comments here that I missed.

**Review by GaryRidgway — APPROVED** [review]
REVIEWED

**Comment by pyrello** [comment]
Thanks, @GaryRidgway!

**Pull request merged**